io/zenoh-transport/src/common/pipeline.rs

```diff
@@ -688,28 +688,62 @@ impl TransmissionPipeline {
             });
         }
 
-        let active = Arc::new(AtomicBool::new(true));
+        let active = Arc::new(TransmissionPipelineStatus {
+            disabled: AtomicBool::new(false),
+            congested: AtomicU8::new(0),
+        });
         let producer = TransmissionPipelineProducer {
             stage_in: stage_in.into_boxed_slice().into(),
-            active: active.clone(),
+            status: active.clone(),
             wait_before_drop: config.wait_before_drop,
             wait_before_close: config.wait_before_close,
         };
         let consumer = TransmissionPipelineConsumer {
             stage_out: stage_out.into_boxed_slice(),
             n_out_r,
-            active,
+            status: active,
         };
 
         (producer, consumer)
     }
 }
 
+struct TransmissionPipelineStatus {
+    // The whole pipeline is enabled or disabled
+    disabled: AtomicBool,
+    // Bitflags to indicate the given priority queue is congested
+    congested: AtomicU8,
```

Comment:
Discussional: in order to reduce contention around this atomic, I'd recommend switching to a set of separate `AtomicBool` and make 64-byte padding (maybe indirect by putting them into `StageIn` etc). The cost of operation for multi-(thread,core,priority) setup should be smaller.

Reply:
I thought about that but increasing the number of atomics also increases the amount of data to be kept in cache. Having one atomic per priority will significantly put more burden on moving data across different cache level I assume, thus reducing performance in any case. I don't have an experimental comparison but I'm not expecting a lot of contention in general.